cni/test/e2e-taint/e2e_taint_test.go

```diff
@@ -0,0 +1,256 @@
+	config, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return
+	}
+	clientset, err = client.NewForConfig(config)
+	return
+}
+
+//read file given location
+func buildDaemonSetFromYAML(fileLocation string) (obj v12.DaemonSet, err error) {
+	reader, err := os.Open(fileLocation)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	obj = v12.DaemonSet{}
+	err = yaml.NewYAMLOrJSONDecoder(reader, 4096).Decode(&obj)
+	return
+}
+func createDaemonset(client *client.Interface, fileLocation string) (daemonset *v12.DaemonSet, err error) {
+	obj, err := buildDaemonSetFromYAML(fileLocation)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	daemonset, err = (*client).AppsV1().DaemonSets(obj.Namespace).Create(context.TODO(), &obj, metav1.CreateOptions{})
+	if err != nil {
+		log.Fatalf(err.Error())
+		return nil, err
+	}
+	return daemonset, err
+}
+
+//build configmap given name and namespace in cmd.ControllerOptions and location
+func buildConfigIfNotExists(clientSet client.Interface, options taint.Options, fileLocation string) (configmap *v1.ConfigMap, err error) {
+	err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Delete(context.TODO(), options.ConfigmapName, metav1.DeleteOptions{})
+	fmt.Println(filepath.Abs("./"))
+	currpath, err := filepath.Abs("./")
+	cmd := exec.Command("kubectl", "create", "configmap", options.ConfigmapName, "-n", options.ConfigmapNamespace, "--from-file="+fmt.Sprintf("%s/%s/", currpath, fileLocation))
+	fmt.Println(cmd.String())
+	err = cmd.Run()
+	err = retry.UntilSuccess(func() error {
+		configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
+		return err
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	configmap, err = clientSet.CoreV1().ConfigMaps(options.ConfigmapNamespace).Get(context.TODO(), options.ConfigmapName, metav1.GetOptions{})
+	return configmap, err
+}
+func waitForConfigMapDeletion(client client.Interface, name, namespace string) error {
+	err := retry.UntilSuccess(func() error {
+		_, err2 := client.CoreV1().ConfigMaps(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err2 == nil {
+			return fmt.Errorf("configmap %s should be deleted", name)
+		}
+		if errors.IsNotFound(err2) {
+			return nil
+		}
+		return err2
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	return err
+}
+func waitForDaemonSetDeletion(client client.Interface, name, namespace string) error {
+	err := retry.UntilSuccess(func() error {
+		_, err2 := client.AppsV1().DaemonSets(namespace).Get(context.TODO(), name, metav1.GetOptions{})
+		if err2 == nil {
+			return fmt.Errorf("configmap %s should be deleted", name)
+		}
+		if errors.IsNotFound(err2) {
+			return nil
+		}
+		return err2
+	}, retry.Converge(1), retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+	return err
+}
+func TestTaintController(t *testing.T) {
```

Comment:
Tests relying on kubernetes must be in the test/integration folder. The expectation is that all other tests can be run anywhere without kubernetes (or even internet) acess

Reply:
moved it to test integration folder